src/machine-learning/machine-learning.ts

```diff
@@ -298,7 +303,9 @@ export class Model {
 export interface TFLiteModel {
   readonly sizeBytes: number;
 
-  readonly gcsTfliteUri: string;
+  // Oneof these two
```

Comment:
Another way to represent this kind of relationships is with union types.

```
interface GcsTFLiteModel {
  sizeBytes: number;
  gcsTFliteUri: string;
}

interface AutoMLModel {
  sizeBytes: number;
  automlModelId: string;
}

type TFLiteModel = GcsTFLiteModel | AutoMLModel
```

You can also take the inheritance approach if you want:

```
interface TFLiteModel {
  sizeBytes: number;
}

interface GcsTFLiteModel extends TFLiteModel {
  gcsTfliteUri: string;
}

interface AutoMLModel extends TFLiteModel {
  autoMlModelId: string;
}
```

Not sure if either of these options pay off in this case. But consider the pros and cons before ruling them out.